### DISCUSSION TITLE
Croatian layout

### DISCUSSION THREAD

**Discussion by lukamrko** [8VIM/8VIM#257] [Language Layout]
I used [Croatian bigram counter](https://github.com/lukamrko/croatianBigramCounter) to generate Croatian bigrams and then I used this data to generate layout for Croatian language using the [8 vim keyboard calculator](https://github.com/Glitchy-Tozier/8vim_keyboard_layout_calculator).  This are results:

`
                                                The top 5 BEST layouts:

      \  x                ▓ /
      ▓ \  z            f /  ▓
        y \  p        m /  g
          v \  s    r /  k
            j \     /  n
                \ /
                / \
            a /     \  o
          l /  e    i \  d
        h /  t        c \  b
      ▓ /  w            u \  q
      /  ▓                ▓ \
oidcbuq- eatlwh-- jsvpyz-x rnmkfg--
OIDCBUQ- EATLWH-- JSVPYZ-X RNMKFG--
─────────────────────────────────────────────> Layout-placing: 1
─────────────────────────────────────────────> Score: 4285096.02    ~95.69 %

      \  x                q /
      ▓ \  b            z /  ▓
        u \  p        m /  g
          k \  r    s /  v
            j \     /  n
                \ /
                / \
            o /     \  a
          l /  e    i \  d
        h /  c        t \  f
      ▓ /  y            w \  ▓
      /  ▓                ▓ \
aidtfw-- eoclyh-- jrkpub-x snmvzgq-
AIDTFW-- EOCLYH-- JRKPUB-X SNMVZGQ-
─────────────────────────────────────────────> Layout-placing: 2
─────────────────────────────────────────────> Score: 4283675.97    ~95.66 %

      \  x                q /
      ▓ \  g            h /  ▓
        y \  p        m /  u
          v \  s    r /  k
            j \     /  n
                \ /
                / \
            a /     \  o
          l /  e    i \  d
        z /  t        c \  b
      ▓ /  w            f \  ▓
      /  ▓                ▓ \
oidcbf-- eatlwz-- jsvpyg-x rnmkhuq-
OIDCBF-- EATLWZ-- JSVPYG-X RNMKHUQ-
─────────────────────────────────────────────> Layout-placing: 3
─────────────────────────────────────────────> Score: 4283472.2    ~95.65 %

      \  x                q /
      ▓ \  g            z /  ▓
        u \  p        m /  y
          k \  r    s /  v
            j \     /  n
                \ /
                / \
            o /     \  a
          l /  e    i \  d
        h /  c        t \  b
      ▓ /  f            w \  ▓
      /  ▓                ▓ \
aidtbw-- eoclfh-- jrkpug-x snmvzyq-
AIDTBW-- EOCLFH-- JRKPUG-X SNMVZYQ-
─────────────────────────────────────────────> Layout-placing: 4
─────────────────────────────────────────────> Score: 4282292.02    ~95.63 %

      \  x                ▓ /
      ▓ \  b            h /  ▓
        y \  p        m /  g
          v \  s    r /  k
            j \     /  n
                \ /
                / \
            a /     \  o
          l /  e    i \  d
        z /  t        c \  f
      ▓ /  w            u \  q
      /  ▓                ▓ \
oidcfuq- eatlwz-- jsvpyb-x rnmkhg--
OIDCFUQ- EATLWZ-- JSVPYB-X RNMKHG--
─────────────────────────────────────────────> Layout-placing: 5
─────────────────────────────────────────────> Score: 4282075.82    ~95.62 %
`

This can probably be improved a little bit if diacritics are taken in account, but for first draft it should be good enough. If some other info is needed fell free to ask.

**Comment by Glitchy-Tozier** [comment]
Could you add the last lines of the evaluation, where the script tells us how much it likes the original layout?

**Reply by lukamrko** [reply]
Here is everything:

     Custom layouts:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
     Example Layout:                                                                                                                                                                                                                                                                
     ghopwx-- abijqryz cdklst-- efmnuv--                                                                                                                                                                                                                                            
     GHOPWX-- ABIJQRYZ CDKLST-- EFMNUV--                                                                                                                                                                                                                                            
     ───────────────────────────────────> Score: 3472475.7    ~77.54 %                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
     Old / original 8VIM layout:                                                                                                                                                                                                                                                    
     nomufv-w eilhkj-- tscdzg-- yabrpxq-                                                                                                                                                                                                                                            
     NOMUFV-W EILHKJ-- TSCDZG-- YABRPXQ-                                                                                                                                                                                                                                            
     ───────────────────────────────────> Score: 3914374.2    ~87.41 %                                                                                                                                                                                                              
     #############                                                                                                                                                                                                 
     General Stats:                                                                                                                                                                                                             
     Time needed for the whole runthrough: 813.06 seconds.                                                                                                                                                                                                                          
     Amount of bigrams that can be written with the letters used in this layout (without factoring in flow or layer-penalty):                                                                                                                                                       
     5455581 out of 5455581  ( ~100.00 %)
